apps/full-site-editing/full-site-editing-plugin/premium-content/index.js

```diff
@@ -33,11 +39,40 @@ const registerBlock = ( block ) => {
 	} );
 };
 
+/**
+ * Appends a "paid" tag to the Premium Content block title if site requires an upgrade.
+ */
+const addPaidBlockFlags = async () => {
+	const membershipsStatus = await apiFetch( { path: '/wpcom/v2/memberships/status' } );
+	const shouldUpgrade = membershipsStatus.should_upgrade_to_access_memberships;
```

Comment:
What happens if the fetch fails?

Reply:
Oh interesting! It adds an inline error to the block insert:

<img width="657" alt="Screen Shot 2020-05-28 at 11 14 52 AM" src="https://user-images.githubusercontent.com/1270189/83177847-97e37a80-a0d4-11ea-825a-f097d778baa1.png">

<img width="862" alt="Screen Shot 2020-05-28 at 11 15 10 AM" src="https://user-images.githubusercontent.com/1270189/83177860-9dd95b80-a0d4-11ea-8ed5-56285b205e32.png">

Reply:
> It adds an inline error to the block insert

I think that's actually caused by [this other request](https://github.com/Automattic/wp-calypso/blob/0f5b4042c7996f88f38376ac501e047bb8a42149/apps/full-site-editing/full-site-editing-plugin/premium-content/blocks/container/edit.js#L215-L218).

> Let's maybe not display paid in the error case and make sure no JS errors are thrown.

Yeah, I like this, so I added some error handling in 008680c to make sure the flag is not added and no JS errors are shown:

<img width="1277" alt="Screen Shot 2020-05-29 at 11 18 50" src="https://user-images.githubusercontent.com/1233880/83243703-97171b00-a19e-11ea-9c2a-463cbb83c5f6.png">